apps/etterlatte-saksbehandling-ui/client/src/shared/hooks/useApiCall.ts

```diff
@@ -3,11 +3,11 @@ import { ApiError, ApiResponse } from '../api/apiClient'
 
 export function useApiCall<T, U>(
   fn: (req: T) => Promise<ApiResponse<U>>
-): [Result<U>, (args: T, onSuccess?: (result: U) => void) => void, () => void] {
+): [Result<U>, (args: T, onSuccess?: (result: U) => void, onError?: (error: any) => void) => void, () => void] {
   const [apiResult, setApiResult] = useState<Result<U>>(initial)
 
   const callFn = React.useCallback(
-    async (args: T, onSuccess?: (result: U) => void) => {
+    async (args: T, onSuccess?: (result: U) => void, onError?: (error: any) => void) => {
```

Comment:
kunde man kanskje typet `error` som `ApiError` istedet for `any`? som regel så prøver vi å bruke `apiClient` til å gjøre api-kall, og da wrappes feilen in i en `ApiError`-struktur.